wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -200,22 +209,74 @@ public void checkCredentials() throws IOException, MediaWikiApiErrorException {
 	 * @return name of the logged in user
 	 */
 	public String getCurrentUser() {
-		return this.username;
+		return username;
+	}
+
+	/**
+	 * Sets the maximum time to wait for a server response once the connection was established, in milliseconds.
+	 * For negative values, no timeout is set.
+	 *
+	 * @see HttpURLConnection#setReadTimeout
+	 */
+	public void setReadTimeout(int timeout) {
+		readTimeout = timeout;
+		if (timeout >= 0) {
+			OkHttpClient.Builder builder = getBuilder();
+			builder.readTimeout(timeout, TimeUnit.MILLISECONDS);
+			client = builder.build();
+		}
 	}
-	
+
+	/**
+	 * Sets the maximum time to wait for when establishing a connection, in milliseconds.
+	 * For negative values, no timeout is set.
+	 *
+	 * @see HttpURLConnection#setConnectTimeout
+	 */
+	public void setConnectTimeout(int timeout) {
+		connectTimeout = timeout;
+		if (timeout >= 0) {
+			OkHttpClient.Builder builder = getBuilder();
+			builder.connectTimeout(timeout, TimeUnit.MILLISECONDS);
+			client = builder.build();
```

Comment:
I'm afraid there is a code logic problem here. If you call `setReadTimeout` then `setConnectTimeout` the read timeout is going to be override because a new client is built with only the connect timeout set.

Reply:
Sorry I didn't notice that.